src/shared/dissect-image.c

```diff
@@ -228,6 +229,31 @@ static int make_partition_devname(
 }
 #endif
 
+#if HAVE_BLKID || HAVE_LIBCRYPTSETUP
+static int dissected_partition_open(DissectedPartition *p) {
+        _cleanup_close_ int fd = -1;
+        const char *node;
+
+        assert(p);
+
+        if (!p->decrypted_node && p->mount_node_fd >= 0)
```

Comment:
why the `!p->decrypted_node` check?

Reply:
First, `dissect_image()` opens all partitions and store their fd into `p->mount_node_fd`.
Then, `dissected_image_decrypt()` opens all decrypted partitions and replaces existing fds.
Hence, `p->mount_node_fd` may be for `p->node`, instead of `p->decrypted_node`.